client/components/data/query-memberships/index.js

```diff
@@ -0,0 +1,61 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestProducts } from 'state/memberships/product-list/actions';
+
+class QueryMemberships extends Component {
+	static propTypes = {
+		productId: PropTypes.number,
+		siteId: PropTypes.number.isRequired,
+		requestProduct: PropTypes.func,
+		requestProducts: PropTypes.func,
+	};
+
+	static defaultProps = {
+		productId: null,
+	};
+
+	UNSAFE_componentWillMount() {
+		this.request( this.props );
+	}
+
+	UNSAFE_componentWillReceiveProps( nextProps ) {
+		if ( nextProps.siteId !== this.props.siteId || nextProps.productId !== this.props.productId ) {
+			this.request( nextProps );
+		}
+	}
```

Comment:
the point of `USAFE_` is for existing code, not new code 😉

Reply:
And that is why it was not under review :)

Reply:
That whole PR has to be refactored once someone actually puts resources on this project. Unfortunately, there are none and the clock is ticking :/

Reply:
But I am writing a huge "what is all this memberships code" post explaining the situation, so hang tight :)

Reply:
> And that is why it was not under review :)

let's not merge it then 👍

Reply:
Recitified in https://github.com/Automattic/wp-calypso/pull/26022